sheerwater_benchmarking/data/stage_iv.py

```diff
@@ -0,0 +1,127 @@
+"""Stage IV precipitation data product."""
+import numpy as np
+import xarray as xr
+from dateutil import parser
+
+from sheerwater_benchmarking.utils import apply_mask, cacheable, clip_region, dask_remote, regrid, roll_and_agg
+
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['year', 'grid'],
+           chunking={'lat': 300, 'lon': 300, 'time': 365})
+def stage_iv_gridded(year, grid):
+    """Regridded Stage IV precipitation data by year.
+
+    Reads data from USGS Geo Data Portal.
+
+    Data was stored in grib format, but was converted to zarr format.
+    https://code.usgs.gov/wma/nhgf/geo-data-portal/stage_iv_precip
+    We are using the zarr produced from that conversion.
+
+    Data is available from 2002-present.
+    """
+    # Open the zarr store
+    store = 'https://usgs.osn.mghpcc.org/mdmf/gdp/stageiv_combined.zarr'
+    ds = xr.open_dataset(store, engine='zarr', chunks={})
+
+    # Select the year
+    ds = ds.sel(time=slice(f"{year}-01-01", f"{year}-12-31"))
+
+    # Ensure consistent naming
+    ds = ds.rename(
+        {'Total_precipitation_surface_1_Hour_Accumulation': 'precip'},
+    )
+
+    # stage iv precipitation is measured in kg/m^2 and hourly
+    # 1 kg/m^2 = 1 mm
+    # to convert to our target units of mm/day, we are summing to daily values
+    ds = ds.resample(time='1D').sum()
+    ds.attrs.update(units='mm/day')
+
+    # Fix coordinates and dimensions
+    # Data is in polar stereographic projection with 2D lat/lon coordinates
+    # Need to interpolate to a regular lat/lon grid before regridding
+
+    # Get min/max bounds of the data
+    lat_min = ds.latitude.min().values
+    lat_max = ds.latitude.max().values
+    lon_min = ds.longitude.min().values
+    lon_max = ds.longitude.max().values
+
+    # Create regular lat/lon coordinates at approximately the same resolution
+    n_lat = ds.dims['y']
+    n_lon = ds.dims['x']
+    new_lats = np.linspace(lat_min, lat_max, n_lat)
+    new_lons = np.linspace(lon_min, lon_max, n_lon)
+
+    # Interpolate to the new regular grid
+    ds = ds.interp(
+        y=new_lats,
```

Comment:
Genevieve and I talked and think that interp might need to use lat/lon instead of y/x here? Not positive though. I think we'll need to test and compare to metview or another trusted regridder. Did this code work?